planner/core/task.go

```diff
@@ -151,18 +151,19 @@ func (t *copTask) finishIndexPlan() {
 	t.indexPlanFinished = true
 	sessVars := t.indexPlan.SCtx().GetSessionVars()
 	// Network cost of transferring rows of index scan to TiDB.
-	t.cst += cnt * sessVars.NetworkFactor * t.tblColHists.GetAvgRowSize(t.indexPlan.SCtx(), t.indexPlan.Schema().Columns, true, false)
+	t.cst += cnt * sessVars.GetNetworkFactor(nil) * t.tblColHists.GetAvgRowSize(t.indexPlan.SCtx(), t.indexPlan.Schema().Columns, true, false)
```

Comment:
For temporary tables, the network factor is 0. Is it possible to pass a `TableInfo`?

Reply:
L161 will recount cost using TableInfo.